readthedocs/config/tests/test_config.py

```diff
@@ -196,6 +196,58 @@ def test_build_config_has_source_file(tmpdir):
     assert build.source_file == os.path.join(base, 'readthedocs.yml')
 
 
+def test_load_non_default(tmpdir):
+    """
+    Load a config file name from non-default path
+
+    Verifies that we can load a custom config path and that an existing default config file is
+    correctly ignored.
+    """
+    non_default_filename = "myconfig.yaml"
+    apply_fs(
+        tmpdir,
+        {
+            non_default_filename: textwrap.dedent(
+                """
+            version: 2
+        """
+            ),
+            ".readthedocs.yaml": "illegal syntax but should not load",
+        },
+    )
+    base = str(tmpdir)
+    with override_settings(DOCROOT=tmpdir):
+        build = load(base, {}, readthedocs_yaml_path="myconfig.yaml")
+    assert isinstance(build, BuildConfigV2)
+    assert build.source_file == os.path.join(base, non_default_filename)
+
+
+def test_load_non_default_with_strange_extension(tmpdir):
+    """
+    Load a config file name from non-default path
+
+    In this version, we verify that we can handle non-yaml extensions
+    because we allow the user to do that.
+    """
```

Comment:
I thought we decided to only support `readthedocs.yaml`, didn't we? I'm not opposed to this, tho.

Reply:
I'll add a comment about this test case, so it's clear what it's doing... it just wants to check that the default is NOT loaded when there's another path specified for `config.load` :)

The validator of the `readthedocs_yaml_path` `CharField` enforces `.readthedocs.yaml`, but the validator is only defined and used on the `CharField`. It will work in the django-admin and on all ModelForms.

I think we're okay to allow any kind of path for internal functions, after all we could change validation over time and still have to support paths from the past (ouch, I bit my _thunge_).